python/cudf/cudf/core/column/categorical.py

```diff
@@ -1417,8 +1417,17 @@ def _get_decategorized_column(self) -> ColumnBase:
 
     def copy(self, deep: bool = True) -> CategoricalColumn:
         if deep:
+
             copied_col = libcudf.copying.copy_column(self)
             copied_cat = libcudf.copying.copy_column(self.dtype._categories)
+            if is_interval_dtype(self.dtype._categories):
+                # avoiding circular import since IntervalColumn
+                # inherits from StructColumn.
+                from cudf.core.column import IntervalColumn
```

Comment:
Just a note that I'm using this full import statement because if I try to use `cudf.core.column.IntervalColumn.from_struct_column` Black complains that the line is too long.